tequilapi/endpoints/transactor.go

```diff
@@ -367,12 +370,38 @@ func (te *transactorEndpoint) SettlementHistory(resp http.ResponseWriter, req *h
 		query.FilterAccountantID(common.HexToAddress(accountantID))
 	}
 
+	page := 1
+	if pageStr := req.URL.Query().Get("page"); pageStr != "" {
+		var err error
+		if page, err = strconv.Atoi(pageStr); err != nil {
+			utils.SendError(resp, err, http.StatusBadRequest)
+			return
+		}
+	}
+
+	pageSize := 50
+	if pageSizeStr := req.URL.Query().Get("page_size"); pageSizeStr != "" {
+		var err error
+		if pageSize, err = strconv.Atoi(pageSizeStr); err != nil {
+			utils.SendError(resp, err, http.StatusBadRequest)
+			return
+		}
+	}
+
 	if err := te.settlementHistoryProvider.Query(query.FetchEntries()); err != nil {
 		utils.SendError(resp, err, http.StatusInternalServerError)
 		return
 	}
 
-	response := contract.NewSettlementListResponse(query.Entries)
+	var settlements []pingpong.SettlementHistoryEntry
+	p := paginator.New(adapter.NewSliceAdapter(query.Entries), pageSize)
```

Comment:
You're not paging in DB so this has no benefit besides sending a partial result back.
You can page in storm DB, e.g.:
```
err = db.Find("Group", "staff", &users, storm.Limit(10), storm.Skip(10))
```

Reply:
It's 2nd place there we depend on this paginator, didn't write bolt adapter for it because:
- more simple to use it like this
- bolt is traversing&reducing all records in your suggested approach anyways